bubuku/features/restart_if_dead.py

```diff
@@ -45,12 +50,19 @@ def __init__(self, broker: BrokerManager, zk: BukuExhibitor):
         super().__init__()
         self.broker = broker
         self.zk = zk
+        self.need_check = True
 
     def check(self) -> Change:
+        if not self.need_check:
+            return None
         if self.broker.is_running_and_registered():
             return None
         _LOG.info('Oops! Broker is dead, triggering restart')
-        return StartBrokerChange(self.broker, self.zk)
+        self.need_check = False
+        return StartBrokerChange(self.broker, self.zk, self.on_check_removed)
+
+    def on_check_removed(self):
+        self.need_check = True
```

Comment:
I do not understand the purpose of this callback method. Could you elaborate on it ?